### PR TITLE
IGNITE-12580: NPE in GridMetricManager.

### PR DESCRIPTION
**Pull request by nizhikov**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/metric/GridMetricManager.java [hunk 1]
```diff
@@ -386,14 +386,19 @@ public void remove(String regName) {
 
         notifyListeners(mreg, metricRegRemoveLsnrs, log);
 
+        DistributedMetaStorage metastorage0 = metastorage;
+
+        if (metastorage0 == null)
```
Comment (agura) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/metric/GridMetricManager.java]: Good catch!

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/LocalWalModeChangeDuringRebalancingSelfTest.java [hunk 2]
```diff
@@ -234,16 +230,6 @@ public void testWalDisabledDuringRebalancing() throws Exception {
         doTestSimple();
     }
 
-    /**
```
Comment (agura) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/LocalWalModeChangeDuringRebalancingSelfTest.java]: Why so many tests are removed? It isn't related with NPE in `GridMetricManager`.

Comment (nizhikov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/LocalWalModeChangeDuringRebalancingSelfTest.java]: The changes in the tests doesn't relate to the fix.
I removed it to run test that reproduces error many times on the TC - https://ci.ignite.apache.org/viewLog.html?buildId=4969452&.

To fix the issue only changes in GridMetricManager required.

If you know the way write the reproducer, please, let me know.

Comment (agura) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/LocalWalModeChangeDuringRebalancingSelfTest.java]: I believe it's impossible to write adequate reproducer except of a lot of test repetition. But I think that configuration should not be deleted on cache stop, only on cache destroy.

**Pull request merged**